hail/src/main/scala/is/hail/methods/IBD.scala

```diff
@@ -304,28 +306,16 @@ object IBD {
     bounded: Boolean = true,
     min: Option[Double] = None,
     max: Option[Double] = None): TableIR = {
-    min.foreach(min => optionCheckInRangeInclusive(0.0, 1.0)("minimum", min))
-    max.foreach(max => optionCheckInRangeInclusive(0.0, 1.0)("maximum", max))
-
     ExecuteContext.scoped { ctx =>
-      val input = Interpret(inputIR, ctx, optimize = true).toMatrixValue(inputIR.typ.colKey)
-      input.requireUniqueSamples("ibd")
-
-      min.liftedZip(max).foreach { case (min, max) =>
-        if (min > max) {
-          fatal(s"minimum must be less than or equal to maximum: ${ min }, ${ max }")
-        }
-      }
-
-      val computeMaf = mafFieldName.map(generateComputeMaf(input, _))
-      val sampleIds = input.stringSampleIds
-
-      TableLiteral(TableValue(ctx, ibdPType, FastIndexedSeq("i", "j"),
-        computeIBDMatrix(input, computeMaf, min, max, sampleIds, bounded)), ctx)
+      val ibd = IBD(mafFieldName, bounded, min, max)
+      val mv = Interpret(inputIR, ctx, optimize = true).toMatrixValue(inputIR.typ.colKey)
+      val tv = ibd.execute(ctx, mv)
+      TableLiteral(tv, ctx)
```

Comment:
The right solution to this change involves deleting `pyApply` entirely -- instead, we would construct a `MatrixToTableApply` IR node in Python containing a serializable representation of the IBD you've written below (which looks totally correct)

Reply:
Gotcha